HesoyamHospital/WebApplication/HospitalSurvey/SurveyValidation.cs

```diff
@@ -1,33 +1,23 @@
-﻿using System;
-using System.Collections.Generic;
-using System.Linq;
-using System.Threading.Tasks;
+﻿using Backend.Model.PatientModel;
 
 namespace WebApplication.HospitalSurvey
 {
     public static class SurveyValidation
     {
         
-        public static bool IsNewSurveyValid(SurveyDTO dto)
+        public static bool IsNewSurveyValid(SurveyDTO dto, Appointment appointment)
         {
-            if( dto == null)
-            {
-                return false;
-            }
-            if(dto.AnswerOne>=1 && dto.AnswerOne<=5 && dto.AnswerTwo >= 1 && dto.AnswerTwo <= 5 &&
+            if( dto == null) return false;
+            if (!appointment.TimeInterval.IsInThePast() || appointment.Canceled || !appointment.AbleToFillOutSurvey) return false;
+            if (dto.AnswerOne>=1 && dto.AnswerOne<=5 && dto.AnswerTwo >= 1 && dto.AnswerTwo <= 5 &&
                 dto.AnswerThree >= 1 && dto.AnswerThree <= 5 && dto.AnswerFour >= 1 && dto.AnswerFour <= 5 &&
```

Comment:
This part of code should definitily be worked on.
The solution appears way too hard coded.
What happens if we want to create a survey which has only 3 answers?

If I'm wrong, please let me know.

Reply:
Apart from that, everything else is great.

Reply:
I agree with you on that one, ir's only that this wasn't focus of my work. Related to surveys, my task was to ensure that no surveys can be filled out for appointments that are not finished. Nevertheless, your suggestion will be taken into consideration by my team for future improvements.